aten/src/ATen/mps/MPSAllocator.h

```diff
@@ -0,0 +1,242 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/Tensor.h>
+#include <ATen/ATen.h>
+#include <ATen/Utils.h>
+#include <torch/library.h>
+
+#include <ATen/mps/MPSDevice.h>
+#include <cstdio>
+#include <mutex>
+#include <unordered_map>
+#include <set>
+#include <utility>
+#include <mach/vm_page_size.h>
+
+#ifdef __OBJC__
+#include <Foundation/Foundation.h>
+#include <Metal/Metal.h>
+#include <Metal/MTLHeap.h>
+#include <MetalPerformanceShaders/MetalPerformanceShaders.h>
+#endif
+
+// NOTE: this header must not be included directly anywhere, except for MPSAllocator.mm
+// Use "at::mps::GetMPSAllocator()" to acquire a handle to MPS Allocator
```

Comment:
if this header can only be included from one place, can we just inline it into MPSAllocator.mm ?

Reply:
Actually that comment was not accurate. We do include it from couple of places. We fixed the comment and added more documentation to it.